lib/table_sync/plugins.rb

```diff
@@ -0,0 +1,80 @@
+# frozen_string_literal: true
+
+# @api public
+# @since 2.2.0
+module TableSync::Plugins
+  require_relative "plugins/registry"
+  require_relative "plugins/access_mixin"
+  require_relative "plugins/abstract"
+
+  # @since 2.2.0
+  @plugin_registry = Registry.new
+  # @since 2.2.0
+  @access_lock = Mutex.new
+
+  class << self
+    # @param plugin_name [Symbol, String]
+    # @return [void]
+    #
+    # @api public
+    # @since 2.2.0
+    def load(plugin_name)
+      thread_safe { plugin_registry[plugin_name].load! }
+    end
+
+    # @return [Array<String>]
+    #
+    # @api public
+    # @since 2.2.0
+    def loaded_plugins
+      thread_safe do
+        # rubocop:disable Style/MultilineBlockChain
+        plugin_registry.select do |_plugin_name, plugin_module|
```

Comment:
Also, why not incapsulate this inside of registry? So it would become something like:
 `thread_safe { plugin_registry.loaded_plugins }`

Reply:
oh :) i said the same below ;D